{
  "package_name": "h2o",
  "patch_versions": [
    "3.46.0.6"
  ],
  "vulnerable_ranges": [
    [
      "3.10.0.3",
      "3.46.0.5"
    ]
  ],
  "cwe": [
    "CWE-400"
  ],
  "tldr": "Affected versions of this package are vulnerable to uncontrolled resource consumption, which could lead to a Denial of Service (DoS). The `run_tool` command in the `rapids` component allows the `main` function of any class under the `water.tools` namespace to be called. Specifically, when the `MojoConvertTool` class is invoked with an invalid argument, it causes the server to crash, potentially resulting in resource exhaustion or disruption of service.",
  "doest_this_affect_me": "You are affected if you are using a version that falls within the vulnerable range.",
  "how_to_fix": "Upgrade the `h2o` library to the patch version.",
  "reporter": "",
  "vulnerable_to": "Uncontrolled Resource Consumption",
  "related_cve_id": "CVE-2020-5979",
  "language": "python",
  "severity_class": "HIGH",
  "aikido_score": 82,
  "changelog": "https://github.com/h2oai/h2o-3/blob/master/Changes.md",
  "published": "2024-11-05",
  "last_modified": "2024-11-05"
}